en/docs/chapter_tree/binary_search_tree.md

```diff
@@ -116,9 +116,9 @@ Given a set of data, we consider using an array or a binary search tree for stor
 | Insert element | $O(1)$         | $O(\log n)$        |
 | Remove element | $O(n)$         | $O(\log n)$        |
 
-In ideal conditions, the binary search tree is "balanced," thus any node can be found within $\log n$ loops.
+Ideally, the binary search tree is "balanced", so that any node can be found within $\log n$ loops.
```

Comment:
optional:
"Ideally, the binary search tree is "balanced", so that any node can be found within $\log n$ loops."
->
"Ideally, the binary search tree is "balanced," allowing any node to be found within $\log n$ loops."